xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/General.h

```diff
@@ -141,5 +141,35 @@ namespace gui
   }
   //--------------------------------------------------------------------------
 
+  //==========================================================================
+  ///
+  /// \ingroup cpp_kodi_gui
+  /// \brief To get hardware specific device context interface
+  ///
+  /// \return                        The currently active device context
+  ///
+  /// \warning This function is only be supported under Windows, on all other
+  /// OS it return `nullptr`!
+  ///
+  /// \note Returned Windows class pointer is `ID3D11DeviceContext1`.
+  ///
+  ///
+  ///-------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// #include <d3d11_1.h>
+  /// ..
+  /// ID3D11DeviceContext1* context = reinterpret_cast<ID3D11DeviceContext1>(kodi::gui::GetD3DContext());
```

Comment:
This line doesn't look right

Reply:
Ups, becomes changed to `GetHWContext` (Was the name before)

Reply:
```suggestion
  /// ID3D11DeviceContext1* context = reinterpret_cast<ID3D11DeviceContext1*>(kodi::gui::GetHWContext());
```
wrong reinterpret_cast's type param :)

Reply:
Also static_cast might suffice?

Reply:
Is updated, yes with static_cast also worked and is changed.